### PR TITLE
Add documentation for devcontiner

### PR DESCRIPTION
**Pull request by ykadowak**
### Description:

With https://github.com/vdaas/vald/pull/2032, our development experience with `devcontainer` has improved. This PR updates the documentation to reflect those changes.

### Versions:

- Go Version: 1.20.3
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 2.0.11

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+1. Change `example/helm/values.yaml` to `dimensions: 784` and `distance_type: l2`.
+2. Run the commands below.
+
+```bash
+# Download the dataset
+make hack/benchmark/assets/dataset/fashion-mnist-784-euclidean.hdf5
+
+# Create k3d cluster
+k3d cluster create vald
```
Comment (hlts2) [inline DEVELOPMENT.md:53]: How about the following? 🤔 :
```suggestion
make k3d/start
```

Comment (ykadowak) [reply DEVELOPMENT.md]: `make k3d/start` creates a lot of pods that are not required for running the E2E test, leading to a log wait until those pods become ready. So I think just creating a plain k3d is better here.

File: docs/contributing/development.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+The command below will run all the unit tests.
+
+```bash
+make test
+```
+
+### E2E tests
+
+The steps below will deploy `Vald` to local `k3d` cluster and run the E2E tests.
```
Comment (vankichi) [inline docs/contributing/development.md:43]:
```suggestion
The steps below will deploy a Vald cluster to the local `k3d` cluster and run the E2E tests.
```

File: docs/contributing/development.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+This is the easiest way to start developing `Vald`. You can just open our [devcontainer.json](https://github.com/vdaas/vald/blob/main/.devcontainer/devcontainer.json) with `VS Code` and go.
+
+### Other
+
+We don't officially have a setup documentation for now, but you can take a look at the [`Dockerfile`](https://github.com/vdaas/vald/blob/main/dockers/dev/Dockerfile).
+That's everything you need to build and test `Vald`, so you can use it as a reference.
+
+> If you would like to use the `Dockerfile` directlly, please note that `docker-in-docker` environment is required to run our E2E tests.
+> In devcontainer, [`VS Code` handles it for us](https://github.com/devcontainers/features/tree/main/src/docker-in-docker).
```
Comment (vankichi) [inline docs/contributing/development.md:29]: To display it as a card view on the official website, you will need to alter the format.
It is a just suggestion, so you do not have to change.

Ref: https://vald.vdaas.org/docs/user-guides/upgrade-cluster/

**Review by vankichi — APPROVED** [review]

File: docs/contributing/development.md (hunk 3 above)
Comment (ykadowak) [reply docs/contributing/development.md:29]: Let's check how it will look like on the web.

**Review by hlts2 — APPROVED** [review]
LGTM!

**Review by kpango — APPROVED** [review]

**Pull request merged**